### PR TITLE
Add validation set in train and pretrain files

### LINKED ISSUE #150: Add validation set and callbacks in pretrain and train scripts
Currently, we only have train accuracy and its not being validated. We can go for a simple validation script or a K fold validation. 
With validation check , we can add callbacks such as early stopping and model checkpoints.

### PR DESCRIPTION
**Pull request by osmarcedron**
Adds the validation_set

Fixes #150.

### CODE REVIEW

File: pretrain.py [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 from autowebcompat import network, utils
 
+SAMPLE_SIZE = 3000
```
Comment (marco-c) [inline pretrain.py]: While I like these changes, moving the constants at the top, they are unrelated to the objective of this PR, so they should be in a separate PR.

**Comment by osmarcedron** [comment]
@marco-c, the changes with respect to the unrelated changes (#167) were addressed, could you please check this PR?

**Comment by osmarcedron** [comment]
Done! @marco-c, check it again please.
